src/github.com/matrix-org/go-neb/services/slackapi/slackapi.go

```diff
@@ -0,0 +1,58 @@
+package slackapi
+
+import (
+	"github.com/matrix-org/go-neb/matrix"
+	"github.com/matrix-org/go-neb/plugin"
+	"github.com/matrix-org/go-neb/types"
+	"net/http"
+	"strings"
+)
+
+type slackAPIService struct {
+	id                 string
+	serviceUserID      string
+	webhookEndpointURL string
+	ClientUserID       string
+	Hooks              map[string]string
+}
+
+func (s *slackAPIService) ServiceUserID() string                 { return s.serviceUserID }
+func (s *slackAPIService) ServiceID() string                     { return s.id }
+func (s *slackAPIService) ServiceType() string                   { return "slackapi" }
+func (s *slackAPIService) PostRegister(oldService types.Service) {}
+func (s *slackAPIService) Register(oldService types.Service, client *matrix.Client) error {
+	return nil
+}
+
+func (s *slackAPIService) Plugin(cli *matrix.Client, roomID string) plugin.Plugin {
+	return plugin.Plugin{}
+}
+
+func (s *slackAPIService) OnReceiveWebhook(w http.ResponseWriter, req *http.Request, cli *matrix.Client) {
+	segments := strings.Split(req.URL.Path, "/")
```

Comment:
You're right in that it cannot happen in its current form. However, changing the path in another file should not cause the entire process to panic and die. Without length checks that's exactly what will happen.